ImageChecks.ps1

```diff
@@ -0,0 +1,167 @@
+        Write-Host "Done extracting"
+    } else {
+        Write-Host "Something went wrong while extracting"
+        exit
+    }
+    Remove-Item $path\*.tar -ErrorAction Stop
+    [String]$imgExtracted = Get-ChildItem $path\*.*
+    return [String]$imgExtracted
+}
+
+function Delete-Extracted {
+    <#
+    .SYNOPSIS
+    Deletes the extracted file
+    #>
+    param($file)
+
+    Write-Host "Deleting $file..."
+    $path = Split-Path -Path $file
+    Remove-Item -r $path -ErrorAction Stop
+}
+
+function Check-Format {
+    <#
+    .SYNOPSIS
+    Checks the format of an image for the given type
+    .DESCRIPTION
+    This function checks the format of an image to see if it corresponds to the given type,
+    by looking at its extension. After this, based on the type it proceeds
+    with further checks, for MAAS checking that the image has curtin,
+    for KVM looking for VirtIO drivers
+    .PARAMETER Type
+    The image type, it will be taken from the config file
+    .PARAMETER Image
+    The image that will be checked
+    #>
+    param($Type, $Image)
+
+    $extension = [IO.Path]::GetExtension($Image)
+    $extension = $extension.split(".")[1]
+
+    if ($Type -eq "Hyper-V") {
+        if ($extension -eq "VHDX") {
+            Write-Host "OK"
+        } else {
+            Write-Host "Wrong format for Hyper-V"
+            exit
+        }
+    } elseif ($Type -eq "MAAS") {
+        if ($extension -eq "raw") {
+            $converted = $Image.split(".")[0]+".vhdx"
+            Write-Host "Converting $Image to $converted..."
+            qemu-img convert -f $extension -O vhdx $Image $converted
+            if ($? -ne "True") {
+                Write-Host "Error while converting"
+                exit
+            }
+            Write-Host "Mounting $converted..."
+            $volume = Mount-VHD -Path $converted -Passthru | Get-Disk | Get-Partition | Get-Volume
+            $volume = $volume.DriveLetter
+            $volume = $volume + ":\"
+            Write-Host "Checking for curtin..."
+            $curtin = ls $volume | Select-String -Quiet "curtin"
+            if ($curtin) {
+                Write-Host "OK"
+            } else {
+                Write-Host "$Image doesn't have curtin"
+            }
+            Dismount-VHD $converted
+            Remove-Item $converted -ErrorAction Stop
+        } else {
+            Write-Host "Wrong format for MAAS"
+            exit
+        }
+    } elseif ($Type -eq "KVM") {
+        if ($extension -eq "raw" -Or $extension -eq "qcow2") {
+            $converted = $Image.split(".")[0]+".vhdx"
+            Write-Host "Converting $Image to $converted..."
+            qemu-img convert -f $extension -O vhdx $Image $converted
+            if ($? -ne "True") {
+                Write-Host "Error while converting"
+                exit
+            }
+            Write-Host "Mounting $converted..."
+            $volume = Mount-VHD -Path $converted -Passthru | Get-Disk | Get-Partition | Get-Volume

```

Comment:
keep the code DRY (do not repeat yourself)

you have the same code at line ~100